packages/rest/index.ts

```diff
@@ -19,6 +19,13 @@ export * from "./src/client/RestConstants";
 export * from "./src/client/RestStandAloneUtils";
 export * from "./src/client/RestClientError";
 
+/* The following line is valid, but tslint barfs on it.
+ * So, we exclude this file in imperative/tslint-packages.json.
+ */
+export * as SessConstants from "./src/session/SessConstants";
```

Comment:
Not second guessing, just curious, do we often use "export ... as" in these files instead of "export * from"?

Reply:
I'm afraid I don't see other index files having something exported with a given name. 
If we change it to `export * from`, it can still be imported as `sessConstants` if desired whenever it's used.
Should we try to avoid giving names to export since it could be a beaking change if we change (for example) from `SessConstants` to `sessConstants` in the future?

Reply:
perhaps, and perhaps this is the cause of the linting failure as well

Reply:
We were using numerous duplicated literals, like "LtpaToken2" throughout the code. I wanted to create a single symbol that was used everywhere (like SessConstants.TOKEN_TYPE_LTPA). I also wanted the ISession interface to avoid using duplicated literals. Instead, specifying a type that can be any one of several symbols:

    type TOKEN_TYPE_CHOICES = typeof TOKEN_TYPE_LTPA | typeof TOKEN_TYPE_JWT |
             typeof TOKEN_TYPE_APIML

I found no way to export type definitions from a Class into the ISession interface. Thus SessConstants is not a class. Each symbol had to be exported individually. See comments at the beginning of SessConstants.ts.

Because the symbols are not within a class, if we use "export * from", we lose the namespace qualifier. Every symbol would have to be referenced just by the name, like TOKEN_TYPE_LTPA. By using "export * as SessConstants" we are able to use expressions like SessConstants.TOKEN_TYPE_LTPA, which I feel is very helpful in finding the value when reading code.

My research on the phrase "export * as SessConstants" within NPM identifies that it is valid syntax, and clearly it compiles and works correctly. Lint does not report that phrase as an error - lint just crashes. Thus, I decided that was a bug in lint.

Despite being different than the other ways that we export, I would lobby that the benefits outweigh the differences.

Reply:
Sorry, I prematurely pushed changes to remove `export ... as` and reverted the lint rule - trying to knock off some of these comments.  

I could:
- revert my recent changes
- convert `SessConstants.ts` to `namespace` so it can be imported in `ISession.ts`
- leave my recent changes (we do export lots of symbols without a namespace e.g. `CoreMessages.ts`)

Side note: we use classes with all `public static readonly` for the purpose of `namespacing`  but is different than using the `namespace` keyword.  

Please let me know

Reply:
At this point, I will settle for anything that compiles. Note, the CLI must also compile successfully. The current errors in the CLI are like this:

packages/rest/src/api/ZosmfRestClient.ts:12:74 - error TS2724: Module '"../../../../../imperative/lib"' has no exported member 'SessConstants'. Did you mean 'RestConstants'?                                   
                                                                                                        
12 import { IImperativeError, Logger, RestClient, TextUtils, RestConstants, SessConstants } from "@zowe/imperative";

Reply:
@zFernand0 , @tjohnsonBCM  - any strong opinion?

Reply:
"what do?" haha

Reply:
i reverted the changes in the meantime so that @gejohnston can build.  whatever changes are recommended, i'll push both at the same time

Reply:
Not a strong opinion, just a comment :yum:

So by doing `export * as SessConstants from <filepath>` we can then import it like `import { SessContants } from <package>`? 

If that's the behavior we want, and not having to do `import * as SessConstants from <package>/file/path` (which we don't usually see very often), I think the "consistent" way is to create a "namespace" (a class in out case) to export the constants that we want out of it.

Reply:
leaving @gejohnston  implementation